### DISCUSSION TITLE
This ٍError is displayed to me : WiFiClientSecureAxTLS.h: No such file or directory

### DISCUSSION THREAD

**Discussion by mahdiEng** [luc-github/ESP3D#686] [Questions]
hi luc-github.
 when I want to upload your code to my esp8266 this error is displayed to me :
WiFiClientSecureAxTLS.h: No such file or directory
 
how can I solve this?

**Comment by luc-github** [comment]
what version do you use ?
How do you compile ?

**Reply by mahdiEng** [reply]
I'm using the latest version .
I'm compiling with arduino IDE . 
and I chose nodemcu 0.9 (esp-12 module) in board setting.
cpu frequency : 160MHz
upload speed : 115200
flash size : 4M(3M SPIFFS)

**Reply by luc-github** [reply]
Latest version of which branch 2.1.X, or 3.0?
Which core of ESP8266 are you using ? check your arduino ide board manager to know the version

**Reply by mahdiEng** [reply]
2.1x brunch.
 my core is LoLin NodeMcu v3.

**Reply by luc-github** [reply]
what esp8266 core in arduino IDE open board manager to know the version
![image](https://user-images.githubusercontent.com/8822552/139667693-11f07ae0-736b-4a33-a61e-1647edc4d253.png)

**Reply by mahdiEng** [reply]
sorry for misunderstanding.
esp8266 core in arduino IDE  is 2.3.0 .

**Reply by luc-github** [reply]
the miminum core to be used with 2.1.X is 2.5.2 and under 3.X
![image](https://user-images.githubusercontent.com/8822552/139668903-cc47a777-2f77-474b-b290-ad9d7acf844e.png)

**Reply by mahdiEng** [reply]
aha thank you .

**Reply by s1der** [reply]
hi luc,
same problem for me with esp8266 cominity v3.0.2 installed
tried with older versions but meeting other errors

**Reply by luc-github** [reply]
> the miminum core to be used with 2.1.X is 2.5.2 and under 3.X ![image](https://user-images.githubusercontent.com/8822552/139668903-cc47a777-2f77-474b-b290-ad9d7acf844e.png)

esp8266 core 3.0.2 is not under 3.X and is not 2.5.2 so it won't work

**Reply by s1der** [reply]
thank you very much for answering. So I think I missed something.. let's read to find what 👍
